### PR TITLE
Add OpenGraph social preview

### PR DESCRIPTION
**Pull request by josh146**
**Context:** When posting links to the docs on social media, there is no social media preview, as there was no [OpenGraph](https://opengraph.xyz) metadata generated by Sphinx.

**Description of the Change:**

- Adds matplotlib and [sphinxext-opengraph](https://github.com/wpilibsuite/sphinxext-opengraph) as doc requirements

- configures `conf.py` to set the OpenGraph information, in particular,

  - the default social media preview image
  - if an image is used on the page, that is used as the preview image

**Benefits:** As above.

To see a preview, copy and paste the RTDs preview into opengraph.xyz:

- [home page](https://www.opengraph.xyz/url/https%3A%2F%2Fxanaduai-pennylane--4765.com.readthedocs.build%2Fen%2F4765%2F)
- [compiling circuits](https://www.opengraph.xyz/url/https%3A%2F%2Fxanaduai-pennylane--4765.com.readthedocs.build%2Fen%2F4765%2Fintroduction%2Fcompiling_circuits.html)

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -45,9 +45,26 @@
...
+    "sphinxext.opengraph",
     "m2r2",
 ]
 
+# opengraph details
+ogp_type = "article"
+# Comment below to activate custom
+# social media cards
+# ogp_use_first_image = False
+# ogp_image = "_static/header-tall.png"
```
Comment (josh146) [inline doc/conf.py:57]:
```suggestion
ogp_use_first_image = False
# ogp_image = "_static/header-tall.png"
```

File: doc/conf.py [hunk 2]
```diff
@@ -45,9 +45,26 @@
...
+ogp_title = "PennyLane Documentation"
+ogp_type = "website"
+ogp_description = "PennyLane is a cross-platform Python library for differentiable programming of quantum computers. Train a quantum computer the same way as a neural network."
+ogp_social_cards = {
+    "image": "_static/logo.png",
+    "enable": True,
+    "site_url": "https://docs.pennylane.ai/",
+    "line_color": "#03b2ff",
+}
+ogp_image = "_static/header-tall.png"
```
Comment (trbromley) [inline doc/conf.py:62]: There is some cutoff with the tall version:

image

Did you try the not-tall version?

**Review by trbromley — APPROVED** [review]
Thanks @josh146! One minor question, but overall looks great!

File: doc/conf.py (hunk 2 above)
Comment (josh146) [reply doc/conf.py:62]: Unfortunately neither seemed to work :( We need a version properly sized for OG, but I think that can always come later

**Comment by Mandrenkov** [comment]
My feedback on this PR is the same as https://github.com/PennyLaneAI/catalyst/pull/344.

File: doc/conf.py [hunk 3]
```diff
@@ -45,9 +45,26 @@
...
     "sphinx_automodapi.automodapi",
     "sphinx_copybutton",
+    "sphinxext.opengraph",
     "m2r2",
 ]
 
+# Open Graph metadata
+ogp_title = "PennyLane Documentation"
+ogp_type = "website"
+ogp_description = "PennyLane is a cross-platform Python library for differentiable programming of quantum computers. Train a quantum computer the same way as a neural network."
```
Comment (josh146) [inline doc/conf.py:55]:
```suggestion
```

File: doc/requirements.txt [hunk 4]
```diff
@@ -33,3 +33,5 @@ requests~=2.28.1
 # navbar and header data to be updated at the source
 pennylane-sphinx-theme
 tomli~=2.0.0 # Drop once minimum Python version is 3.11
+sphinxext-opengraph
+matplotlib
```
Comment (josh146) [inline doc/requirements.txt:37]:
```suggestion
sphinxext-opengraph==0.9.0
matplotlib==3.8.0
```

File: doc/requirements.txt [hunk 5]
```diff
@@ -33,3 +33,5 @@ requests~=2.28.1
 # navbar and header data to be updated at the source
 pennylane-sphinx-theme
 tomli~=2.0.0 # Drop once minimum Python version is 3.11
+sphinxext-opengraph==0.9.0
```
Comment (josh146) [inline doc/requirements.txt:36]:
```suggestion
sphinxext-opengraph==0.6.3. # Note: latest version 0.9.0 requires Sphinx >=4.0
```

File: doc/requirements.txt [hunk 6]
```diff
@@ -33,3 +33,5 @@ requests~=2.28.1
 # navbar and header data to be updated at the source
 pennylane-sphinx-theme
 tomli~=2.0.0 # Drop once minimum Python version is 3.11
+sphinxext-opengraph==0.6.3 # Note: latest version 0.9.0 requires Sphinx >=4.0
```
Comment (Mandrenkov) [inline doc/requirements.txt:36]: **Question:** Is there something holding us back from upgrading to Sphinx v4?

**Review by Mandrenkov — APPROVED** [review]

File: doc/requirements.txt (hunk 6 above)
Comment (josh146) [reply doc/requirements.txt:36]: No idea, I just didn't want to tackle that in this PR 😆

**Pull request merged**